### PR TITLE
Polishes Kafka Docker image more

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This primarily mentions why relative dir is in use (to make it easy to change).

### CODE REVIEW

File: docker/collector/kafka/Dockerfile [hunk 1]
```diff
@@ -26,17 +25,19 @@ RUN /tmp/install && rm /tmp/install
 FROM openzipkin/jre-full:14.0.2-14.29.23
 LABEL MAINTAINER Zipkin "https://zipkin.io/"
 
-# add HEALTHCHECK and ENTRYPOINT scripts
+# Add HEALTHCHECK and ENTRYPOINT scripts
 COPY docker/collector/kafka/docker-bin/* /usr/local/bin/
 
-# setup non-root user and its home dir
+# All Kafka content including binaries logs writes under WORKDIR
```
Comment (codefromthecrypt) [inline docker/collector/kafka/Dockerfile]:
```suggestion
# All Kafka content including binaries and logs write under WORKDIR
```

File: docker/collector/kafka/Dockerfile [hunk 2]
```diff
@@ -26,17 +25,19 @@ RUN /tmp/install && rm /tmp/install
...
 COPY docker/collector/kafka/docker-bin/* /usr/local/bin/
 
-# setup non-root user and its home dir
+# All Kafka content including binaries and logs write under WORKDIR
+WORKDIR /kafka
+
+# Ensure the process doesn't run as root
 ARG USER=kafka
-WORKDIR /${USER}
-RUN adduser -g '' -h /${USER} -D ${USER}
```
Comment (codefromthecrypt) [inline docker/collector/kafka/Dockerfile:35]: note: the workdir is now decoupled from the convention

File: docker/collector/kafka/docker-bin/docker-entrypoint [hunk 3]
```diff
@@ -14,10 +14,13 @@
...
 
 # ENTRYPOINT script that starts ZooKeeper and then Kafka
+#
+# This intentionally locates config using the current working directory, in order to consolidate
+# Dockerfile instructions to WORKDIR
 set -eu
 
 echo Starting ZooKeeper
-bin/kafka-run-class.sh -Dlog4j.configuration=file:config/log4j.properties org.apache.zookeeper.server.quorum.QuorumPeerMain config/zookeeper.properties &
+bin/kafka-run-class.sh -Dlog4j.configuration=file:./config/log4j.properties org.apache.zookeeper.server.quorum.QuorumPeerMain ./config/zookeeper.properties &
```
Comment (codefromthecrypt) [inline docker/collector/kafka/docker-bin/docker-entrypoint]: clears a minor log warning

File: docker/collector/kafka/Dockerfile [hunk 4]
```diff
@@ -26,17 +25,19 @@ RUN /tmp/install && rm /tmp/install
 FROM openzipkin/jre-full:14.0.2-14.29.23
 LABEL MAINTAINER Zipkin "https://zipkin.io/"
 
-# add HEALTHCHECK and ENTRYPOINT scripts
+# Add HEALTHCHECK and ENTRYPOINT scripts
```
Comment (codefromthecrypt) [inline docker/collector/kafka/Dockerfile]:
```suggestion
# Add HEALTHCHECK and ENTRYPOINT scripts into the default search path
```

File: docker/collector/kafka/Dockerfile [hunk 5]
```diff
@@ -26,17 +25,19 @@ RUN /tmp/install && rm /tmp/install
...
+# Ensure the process doesn't run as root
 ARG USER=kafka
-WORKDIR /${USER}
-RUN adduser -g '' -h /${USER} -D ${USER}
+RUN adduser -g '' -h  ${PWD} -D ${USER}
 USER ${USER}
 
-# copy content we installed earlier into the home dir
-COPY --from=install --chown=${USER} /install /${USER}
+# Copy content we installed earlier
```
Comment (codefromthecrypt) [inline docker/collector/kafka/Dockerfile]:
```suggestion
# Copy binaries and config we installed earlier
```

File: docker/collector/kafka/install [hunk 6]
```diff
@@ -13,16 +13,18 @@
...
 
 echo "*** Installing Kafka and dependencies"
 APACHE_MIRROR=$(curl --stderr /dev/null https://www.apache.org/dyn/closer.cgi\?as_json\=1 | sed -n '/preferred/s/.*"\(.*\)"/\1/gp')
 
-# download kafka binaries
-curl -sSL $APACHE_MIRROR/kafka/$KAFKA_VERSION/kafka_$SCALA_VERSION-$KAFKA_VERSION.tgz | tar xz
-mv kafka_$SCALA_VERSION-$KAFKA_VERSION/* .
+# Download scripts and config for Kafka and ZooKeeper, but not for Connect
+curl -sSL $APACHE_MIRROR/kafka/$KAFKA_VERSION/kafka_$SCALA_VERSION-$KAFKA_VERSION.tgz | tar xvz \
+  --wildcards --strip=1 --exclude=connect* */bin/zookeeper-* */bin/kafka-* */config
```
Comment (codefromthecrypt) [inline docker/collector/kafka/install:26]: the power of GNU tar :D

**Review by anuraaga — APPROVED** [review]

**Review by jeqo — APPROVED** [review]
LGTM 👍

**Comment by codefromthecrypt** [comment]
thanks I have slight more polish nothing very exciting to add :D

**Comment by codefromthecrypt** [comment]
final polish notably gets rid of cygwin stuff.

```bash
$ docker run --rm openzipkin/zipkin-kafka:test
Starting ZooKeeper
[2020-08-30 12:14:06,453] WARN Either no config or no quorum defined in config, running  in standalone mode (org.apache.zookeeper.server.quorum.QuorumPeerMain)
Starting Kafka
[2020-08-30 12:14:08,215] WARN No meta.properties file under dir /opt/kafka/./data/kafka/meta.properties (kafka.server.BrokerMetadataCheckpoint)
```

@jeqo @jorgheymans if you have any ideas to get rid of these two warnings, let's do it as then we have.. no warnings!

fyi here's docker ps
```bash
CONTAINER ID        IMAGE                                  COMMAND                  CREATED             STATUS                   PORTS                                         NAMES
8f9af873e500        openzipkin/zipkin-kafka:test           "start-kafka-zookeep…"   8 seconds ago       Up 7 seconds (healthy)   2181/tcp, 9092/tcp, 19092/tcp                 gracious_archimedes
```

**Comment by codefromthecrypt** [comment]
fyi on the last 2 warnings.. I don't know the setting in ZK to explicitly say you are standalone.. I'm not sure if the warning is just warning that you are that vs lack of a clarifying setting.

the other I currently assume is just saying you are starting a new Kafka. The ID in it seems auto-generated so not sure we should do anything about it.
```bash
$ docker exec -it 9ea8f5613790  /bin/sh
~ $ cat /opt/kafka/./data/kafka/meta.properties
#
#Sun Aug 30 12:17:18 GMT 2020
broker.id=0
version=0
cluster.id=NIjql19sRYOuHAtdiyLsQQ
```

**Comment by codefromthecrypt** [comment]
@eirslett have you noticed this before from github actions? I am not having anything like this locally..

```
Run ./mvnw -q --batch-mode -DskipTests --also-make -pl zipkin-server clean install
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.10.0:install-node-and-npm (install node and npm) on project zipkin-lens: Could not download Node.js: Got error code 522 from the server. -> [Help 1]
```

**Comment by jeqo** [comment]
@adriancole zookeeper log is unavoidable: 

https://github.com/apache/zookeeper/blob/e91455c1e3c50405666cd8afad71d99dceb7b340/zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/QuorumPeerMain.java#L138-L140 

and meta.properties file is generated when broker joins the cluster, using an auto-generated cluster id:

```
cat /opt/kafka/./data/kafka/meta.properties
#
#Mon Aug 31 12:46:30 GMT 2020
broker.id=0
version=0
cluster.id=2eUVIKyFQM2YWzayeONBzw
```

**Comment by codefromthecrypt** [comment]
ok all done. thanks so much for the feedback @anuraaga and @jeqo We can clean the other images similarly.

**Pull request merged**